cmd/gateway/s3x/crdt_dagsyncer.go

```diff
@@ -0,0 +1,122 @@
+package s3x
+
+import (
+	"context"
+
+	pb "github.com/RTradeLtd/TxPB/v3/go"
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-datastore"
+	ipld "github.com/ipfs/go-ipld-format"
+	"github.com/ipfs/go-merkledag"
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+)
+
```

Comment:
It might be a good idea to add a compile check to ensure we satisfy the interface

```
var (
    _ ipld.DAGService = (*crdtDAGSyncer)(nil)
)
```

Reply:
This check already happens at compile time in `gateway-s3x.go`
``` go
func (g *TEMX) newCrdtLedgerStore(ctx context.Context, dag pb.NodeAPIClient, pub pb.PubSubAPIClient) (*ledgerStore, error) {
	...
	crdtds, err := crdt.New(store, datastore.NewKey("crdt"), dagSyncer, pubsubBC, opts)
```

Reply:
ah ok